src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -454,6 +527,54 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void HandleSearchResultFailures(Execution::Context& context)
+    {
+        const auto& searchResult = context.Get<Execution::Data::SearchResult>();
+
+        if (!searchResult.Failures.empty())
+        {
+            if (WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::TreatSourceFailuresAsWarning))
+            {
+                auto warn = context.Reporter.Warn();
+                for (const auto& failure : searchResult.Failures)
+                {
+                    warn << Resource::String::SearchFailureWarning << ' ' << failure.Source->GetDetails().Name << std::endl;
+                }
+            }
+            else
+            {
+                HRESULT overallHR = S_OK;
+                auto error = context.Reporter.Error();
+                for (const auto& failure : searchResult.Failures)
+                {
+                    error << Resource::String::SearchFailureError << ' ' << failure.Source->GetDetails().Name << std::endl;
+                    HRESULT failureHR = HandleException(context, failure.Exception);
+
+                    // Just take first failure for now

```

Comment:
>Just take first failure for now [](http://example.com/codeflow?start=23&length=31)

Should we have a new hr for this instead of taking the first one?

Reply:
I don't want to create a new failure bucket that is "multiple failures" yet.